app/controllers/v0/in_progress_forms_controller.rb

```diff
@@ -32,5 +33,16 @@ def destroy
       form.destroy
       render json: form
     end
+
+    def proxy_add
+      if params[:id] == '21-526EZ'
+        if @current_user.participant_id.nil?
```

Comment:
Is this meant to be the check for if the user is in MVI?  Do we know for sure that the user isn't in MVI with a blank `participant_id`? 

We're only querying MVI for LOA3 users so do we know at this point that the user is LOA3?   We don't want to add non-identity verified users to MVI, do we?

Reply:
Good question. The form 526 flow is gated to log in on the front end AND the service list for their user object. The authorization verifies that the necessary information for the user exists which, I believe, includes them being `user.loa3?`.

But, even if they managed to attempt an add proxy call to MVI, if they dont exist in MVI, it would fail. This call isnt creating an MVI entry but rather, creating entries in CORP and/or BIRLS depending on whether the veteran exists in MVI already. The add proxy call _requires_ an orchestrated search to be made to MVI *before* an add proxy call is made. This accomplishes two things... 1) It creates a timestamped entry in mvi which is important because the proxy add call is time limited to ~90 seconds to the search call that preceded it and 2) the orchestrated search call retrieves a `search_token` var that needs to be included in the proxy call that is made.

Reply:
Long story short, it doesnt check for mvi existence, it just checks if the veteran has an entry in the CORP database

Reply:
Also, the mvi profile isnt accessible at all UNLESS the user is loa3 certified to begin with.

Reply:
Oh, I should add, the auth gating for the service checks our form526 policy. This policy verifies the existence of an `edipi`. The user object, when accessing `edipi` checks whether the user is `loa3?`

Reply:
@annaswims RE: "Do we know for sure that the user isn't in MVI with a blank participant_id?"  

Great question.  When I was writing that part I looked for evidence that the participant and birls ids were `nil` when "empty/blank/none" and I found the sanitize methods for the mvi profile construction.  https://github.com/department-of-veterans-affairs/vets-api/blob/d238959fcd3e040698e35faf41410b1992b9357e/lib/mvi/responses/profile_parser.rb#L164
My reading is that 1. we expect empty/blank/none values to be represented as `nil` in the data source and 2. if the data source value contains only non-digit characters we transform that to `nil` (the return value of an unsuccessful `match`.) https://ruby-doc.org/core-2.5.1/Regexp.html